src/main/java/com/puppycrawl/tools/checkstyle/SarifLogger.java

```diff
@@ -171,7 +243,10 @@ public void auditStarted(AuditEvent event) {
     @Override
     public void auditFinished(AuditEvent event) {
         String rendered = replaceVersionString(report);
-        rendered = rendered.replace(RESULTS_PLACEHOLDER, String.join(",\n", results));
+        rendered = rendered.replace(
+                RESULTS_PLACEHOLDER,
+                String.join(COMMA_LINE_SEPARATOR, results));
+        rendered = rendered.replace(RULES_PLACEHOLDER, generateRulesJson());
```

Comment:
```suggestion
        rendered = rendered.replace(RULES_PLACEHOLDER, generateRulesJson());
```

no reassign. thats coupling. better cohesion make direct usage.